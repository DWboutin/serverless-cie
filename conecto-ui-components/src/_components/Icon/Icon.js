import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import To from '../Logo/Logos/To'
import Menu from './Icons/Menu'
import Alert from './Icons/Alert'
import ArrowDown from './Icons/ArrowDown'
import Bookmark from './Icons/Bookmark'
import Check from './Icons/Check'
import Close from './Icons/Close'
import Cog from './Icons/Cog'
import Concurrents from './Icons/Concurrents'
import Email from './Icons/Email'
import Facebook from './Icons/Facebook'
import Flag from './Icons/Flag'
import ColoredFlag from './Icons/ColoredFlag'
import Marker from './Icons/Marker'
import Notification from './Icons/Notification'
import Phone from './Icons/Phone'
import Search from './Icons/Search'
import SMS from './Icons/SMS'
import ThumbUp from './Icons/ThumbUp'
import User from './Icons/User'

import CCAmex from './Icons/CCAmex'
import CCDefault from './Icons/CCDefault'
import CCDinersClub from './Icons/CCDinersClub'
import CCDiscover from './Icons/CCDiscover'
import CCJCB from './Icons/CCJCB'
import CCMastercard from './Icons/CCMastercard'
import CCVisa from './Icons/CCVisa'

class Icon extends React.Component {
  constructor(props) {
    super(props)

    this.classesPrefix = 'to-icon'
  }

  iconDefinition(type) {
    switch (type) {
      case 'alert':
        return Alert
      case 'arrow-down':
        return ArrowDown
      case 'bookmark':
        return Bookmark
      case 'check':
        return Check
      case 'close':
        return Close
      case 'cog':
        return Cog
      case 'concurrents':
        return Concurrents
      case 'email':
        return Email
      case 'facebook':
        return Facebook
      case 'flag':
        return Flag
      case 'colored-flag':
        return ColoredFlag
      case 'marker':
        return Marker
      case 'menu':
        return Menu
      case 'notification':
        return Notification
      case 'phone':
        return Phone
      case 'search':
        return Search
      case 'sms':
        return SMS
      case 'thumb-up':
        return ThumbUp
      case 'user':
        return User
      case 'cc-amex':
        return CCAmex
      case 'cc-default':
        return CCDefault
      case 'cc-dinersclub':
        return CCDinersClub
      case 'cc-discover':
        return CCDiscover
      case 'cc-jcb':
        return CCJCB
      case 'cc-mastercard':
        return CCMastercard
      case 'cc-visa':
        return CCVisa
      case 'conecto':
      default:
        return To
    }
  }

  render() {
    const { className, type } = this.props
    const classes = classNames({
      [this.classesPrefix]: true,
      [`${className}`]: className,
    })
    const iconToRender = this.iconDefinition(type)

    return (
      <i className={classes} aria-label={`icon: ${type}`}>
        {React.createElement(iconToRender)}
      </i>
    )
  }
}

Icon.defaultProps = {
  type: 'conecto',
}

const IconTypePropTypes = PropTypes.oneOf([
  'alert',
  'arrow-down',
  'bookmark',
  'check',
  'close',
  'cog',
  'concurrents',
  'email',
  'facebook',
  'flag',
  'colored-flag',
  'marker',
  'menu',
  'notification',
  'phone',
  'search',
  'sms',
  'thumb-up',
  'user',
  'cc-amex',
  'cc-default',
  'cc-dinersclub',
  'cc-discover',
  'cc-jcb',
  'cc-mastercard',
  'cc-visa',
  'conecto',
])

Icon.propTypes = {
  type: IconTypePropTypes,
  className: PropTypes.string,
}

export { Icon as default, IconTypePropTypes }
