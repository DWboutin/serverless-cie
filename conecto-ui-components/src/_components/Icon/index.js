import Icon, { IconTypePropTypes } from './Icon'

export { Icon as default, IconTypePropTypes }
