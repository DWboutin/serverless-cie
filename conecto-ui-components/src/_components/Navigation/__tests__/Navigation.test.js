import React from 'react'
import { shallow, mount } from 'enzyme'

import Icon from '../../Icon'
import Facebook from '../../Icon/Icons/Facebook'

global.console = {
  error: jest.fn(),
}

describe('Icon', () => {
  describe('SVG components', () => {
    it('Facebook SVG component should match snapshot', () => {
      const wrapper = shallow(<Facebook />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Default', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should contain the added classes', () => {
      const wrapper = shallow(<Icon className="test-class-1 test-class-2" />)

      expect(wrapper.hasClass('test-class-1')).toEqual(true)
      expect(wrapper.hasClass('test-class-2')).toEqual(true)
    })

    it('should render the short logo is default icon', () => {
      const wrapper = mount(<Icon />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-logo-to')).toEqual(true)

      wrapper.unmount()
    })
  })

  describe('Facebook', () => {
    it('should match snapshot', () => {
      const wrapper = shallow(<Icon type="facebook" />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should have the prefix class', () => {
      const wrapper = shallow(<Icon type="facebook" />)

      expect(wrapper.instance().classesPrefix).toEqual('to-icon')
      expect(wrapper.hasClass(wrapper.instance().classesPrefix)).toEqual(true)
    })

    it('should render the Facebook icon', () => {
      const wrapper = mount(<Icon type="facebook" />)

      expect(wrapper.find('svg').length).toEqual(1)
      expect(wrapper.find('svg').hasClass('to-icon-facebook')).toEqual(true)

      wrapper.unmount()
    })
  })

  it('should console.error when passing a number to type prop', () => {
    shallow(<Icon type={123}>Test</Icon>)

    expect(global.console.error).toBeCalled()
  })

  it('should console.error when not passing a good value to type prop', () => {
    shallow(<Icon type="test">Test</Icon>)

    expect(global.console.error).toBeCalled()
  })
})
