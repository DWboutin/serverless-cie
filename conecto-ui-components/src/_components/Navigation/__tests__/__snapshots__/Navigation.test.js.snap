// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Default should match snapshot 1`] = `
<i
  aria-label="icon: conecto"
  className="to-icon"
>
  <To />
</i>
`;

exports[`Icon Facebook should match snapshot 1`] = `
<i
  aria-label="icon: facebook"
  className="to-icon"
>
  <Facebook />
</i>
`;

exports[`Icon SVG components Facebook SVG component should match snapshot 1`] = `
<svg
  className="to-icon-facebook"
  height="13"
  viewBox="0 0 13 13"
  width="13"
>
  <path
    d="M12.9078 0.737589V12.2624C12.9078 12.6312 12.6312 13 12.1702 13H8.94326V7.92908H10.6028L10.8794 5.99291H8.94326V4.70213C8.94326 4.14894 9.12766 3.78014 9.95745 3.78014H10.9716V2.02837C10.7872 2.02837 10.1418 1.93617 9.49645 1.93617C8.02128 1.93617 7.00709 2.85816 7.00709 4.51773V5.99291H5.25532V7.92908H6.91489V12.9078H0.737589C0.276596 12.9078 0 12.539 0 12.1702V0.737589C0 0.276596 0.276596 0 0.737589 0H12.2624C12.6312 0 12.9078 0.276596 12.9078 0.737589Z"
    fill="#572D2D"
  />
</svg>
`;
